### PR TITLE
Align make_private_with_epsilon with make_private

### LINKED ISSUE #467: mismatch between the documentation and the behaviour of `make_private_with_epsilon`
`PrivacyEngine.make_private_with_epsilon` does  not behave as explained by the documentation. In particular, `poisson_sampling` can not passed as an argument to this method. I think that the error is due to calling `get_noise_multiplier` (in accountants/utils.py) with `**kwargs`, but `RDPAccountant.get_epsilon` (in `rdp.py`) and  `GaussianAccountant.get_epsilon` (in `gdp.py`) only accepts a fixed number of arguments. Also, I think that the signature of GaussianAccountant.get_epsilon should be `get_epsilon(self, delta: float, poisson_sampling: bool = True) -> float`

### PR DESCRIPTION
**Pull request by alexandresablayrolles**
Fixes #467

### CODE REVIEW

**Review by ffuuugor — APPROVED** [review]
Great stuff, thanks!

Accepting to unblock - please make sure tests pass before landing, you've put "max_grad_norm" in there twice

**Review by karthikprasad — APPROVED** [review]

**Pull request closed without merging**
